File: Lista_01/multimetro.java
// Nome: Davi Roriz Oliveira
// Matrícula: 202210342
public class multimetro {
    public static void main(String[] args) {
        float[] valor = { 5, 5.2f, 5.1f, 5, 5.2f }; // Declaração de valores para teste de multímetro
        for (int a = 0; a < valor.length; a++) { // Laço for para printar cada elemento do vetor de valores
            System.out.println("Valor[" + (a + 1) + "]: " + valor[a]);
        }
        System.out.println("Média: " + calcula_media(valor)); // Print da média
        System.out.println("Desvio padrão: " + desvio(valor)); // Print do desvio padrão
        System.out.println();
        if (desvio(valor) > 0.1 * calcula_media(valor)) { // Teste para saber se o multímetro está funcionado
            System.out.println("O multimetro esta com problemas!");
        } else {
            System.out.println("O multimetro esta funcionando!");
        }
    }

    public static float calcula_media(float[] vetor) { // Faz a média dos valores do vetor de testes
        float soma = 0;
        for (int aux = 0; aux < vetor.length; aux++) {
            soma = soma + vetor[aux];
        }
        return soma / vetor.length;
    }

    public static float desvio(float[] vetor) { // Faz o desvio padrão dos valores do vetor de testes
        float media = calcula_media(vetor); // Faz a chamada da média para ser implementada na fórmula do desvio padrão
        float result = 0;
        for (int aux = 0; aux < vetor.length; aux++) {
            result = result + (vetor[aux] - media) * (vetor[aux] - media);
        }
        return (float) Math.sqrt(result / vetor.length);
    }
}

File: multimetro.py
# Nome: Davi Roriz Oliveira
# Matrícula: 202210342

import math

def calcula_media(vetor): # Função que faz o cálculo da média dos valores
    soma = 0
    for aux in range(len(vetor)):
        soma = soma + vetor[aux]
    return soma / len(vetor)
def desvio(vetor): # Função que faz o cálculo do desvio padrão dos valores
    media = calcula_media(vetor)
    result = 0
    for aux in range(len(vetor)):
        result = result + (vetor[aux] - media) * (vetor[aux] - media)
    return math.sqrt(result / len(vetor))

valor = [5, 5.2, 5.1, 5, 5.2] # Declaração de valores para teste de multímetro
for aux in range(len(valor)): # Laço for para printar cada elemento do vetor de valores
    print(f"Valor[{aux + 1}]: {valor[aux]}")
print(f"Média: {calcula_media(valor)}") # Print da média
print(f"Desvio padrão: {desvio(valor)}") # Print do desvio padrão
print()
if desvio(valor) > 0.1 * calcula_media(valor): # Teste para saber se o multímetro está funcionado
    print("O multímetro está com problemas!")
else:
    print("O multímetro está funcionando!")